src/environment.rs

```diff
@@ -95,18 +97,49 @@ pub struct Environment<E> {
 }
 
 impl<E: InnerEnv + 'static> Environment<E> {
-    /// Initialize the `Environment`
+    /// Create new `Environment`
+    ///
+    /// This requires access to a `wl_display` attached to the `event_queue`.
+    /// You also need to provide an instance of the inner environment type declared
+    /// using the [`environment!`](../macro.environment.html) macro.
+    ///
+    /// If you instead used the [`default_environment!`](../macro.default_environment.html), then
+    /// you need to initialize your `Environment` using the
+    /// [`new_default_environment!`](../macro.new_default_environment.html) macro.
+    ///
+    /// `std::io::Error` could be returned if initial roundtrips to the server failed.
+    /// This call may block when doing initial roundtrips, but those blocks could only be due to
+    /// server bugs.
```

Comment:
*Indefinite blocking* can only be caused by server bug, a short blocking while the exchanges are done is expected.